sign_test.go

```diff
@@ -67,6 +69,13 @@ func TestCalulateSigHash(t *testing.T) {
 		r := bytes.NewReader(raw)
 		msgTx := wire.NewMsgTx(1)
 		msgTx.BtcDecode(r, 1, wire.BaseEncoding)
+
+		signerMap := make(map[wire.OutPoint]*wire.TxOut)
```

Comment:
I think you can do `prevOutput := txscript.NewCannedPrevOutputFetcher(prevScript, v.Inputs[idx].Value)` or even `prevOutput := new(txscript.CannedPrevOutputFetcher)`. Both seem to work and are simpler.